### PR TITLE
[@torchlight-api/torchlight-cli] Add type definitions

### PR DESCRIPTION
**Pull request by dominiccarrington**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by peterblazejewicz — APPROVED** [review]
we are advising to add strict optional properties:
https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes
for better interior when released in modern TS

LGTM!

File: types/torchlight-api__torchlight-cli/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    highlight(blocks: Block[]): Promise<Block[]>;
+
+    fan(blocks: Block[]): Promise<Block[]>;
+
+    request(blocks: Block[]): Promise<TorchlightApiResponse["blocks"]>;
+}
+
+interface BlockOptions {
+    id?: string;
+    theme?: TorchlightTheme;
```
Comment (peterblazejewicz) [inline types/torchlight-api__torchlight-cli/index.d.ts:125]:
```suggestion
    id?: string | undefined;
    theme?: TorchlightTheme | undefined;
```
sorry for confusion, I meant here, not within config (DT does not enforce strict notions, just advice to write declaration files in a way compatible with optional settings in tsconfig)

File: types/torchlight-api__torchlight-cli/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        "module": "node16",
+        "lib": ["es6"],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "exactOptionalPropertyTypes": true
```
Comment (peterblazejewicz) [inline types/torchlight-api__torchlight-cli/tsconfig.json:12]:
```suggestion
        "forceConsistentCasingInFileNames": true
```

**Comment by dominiccarrington** [comment]
Thank you for your help @peterblazejewicz, apologies for the misunderstanding

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@dominiccarrington thanks!

**Comment by dominiccarrington** [comment]
Ready to merge

**Pull request merged**